### PR TITLE
Move test to github actions.

### PR DESCRIPTION
**Pull request by godofredoc**
We are moving tests to actions to reduce the maintenance effort.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+name: Tests
```
Comment (Hixie) [inline .github/workflows/tests.yml:1]: do we have any documentation anywhere on how this file works? and if so, could we add a link to it at the top of this file

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        include:
+          - name: tests-linux-0
+            shards: 3
+            shard-index: 0
+          - name: tests-linux-1
+            shards: 3
+            shard-index: 1
+          - name: tests-linux-2
+            shards: 3
+            shard-index: 2
```
Comment (Hixie) [inline .github/workflows/tests.yml:26]: this seems to result in quite ugly shard names like `Tests / test_linux (tests-linux-0, 3, 0) (pull_request)`, any chance we can make it give it pretty names like just `Tests for Linux, shard 0 of 3` or something?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  pull_request:
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  test_linux:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
```
Comment (Hixie) [inline .github/workflows/tests.yml:15]: what's the opposite of "fail-fast"? (and what's the default?)

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      matrix:
+        include:
+          - name: tests-macos-0
+            shards: 2 
+            shard-index: 0
+          - name: tests-macos-1
+            shards: 2
+            shard-index: 1
```
Comment (Hixie) [inline .github/workflows/tests.yml:75]: i worry that this method of sharding is going to mean we end up missing shards one day and not noticing (like if someone just increases the "shards" number but doesn't add a new "shard-index" line). Is there any way to guarantee that all the shards are present?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (godofredoc) [reply .github/workflows/tests.yml:1]: Added a reference to the actions documentation.

File: .github/workflows/tests.yml (hunk 4 above)
Comment (godofredoc) [reply .github/workflows/tests.yml:75]: From a quick look to the docs I didn't find anything that could be used to calculate indexes programmatically before actions read the file and triggers the builds.

**Review by ricardoamador — APPROVED** [review]
LGTM

File: .github/workflows/tests.yml (hunk 3 above)
Comment (Hixie) [reply .github/workflows/tests.yml:18]: if the default is true, should be omit this line? having a line that sets the value to the default is generally misleading (people will assume it's doing something intentional)

Comment (godofredoc) [reply .github/workflows/tests.yml:18]: Sounds good, I'll send a fix!

**Pull request merged**